services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImpl.java

```diff
@@ -422,4 +548,41 @@ protected void validateClientVerificationStatus(String clientId) throws NotFound
 			throw new OAuthClientNotVerifiedException("The OAuth client (" + clientId + ") is not verified.");
 		}
 	}
+
+	@WriteTransaction
+	@Override
+	public void revokeToken(OAuthTokenRevocationRequest revocationRequest) {
+		switch (revocationRequest.getToken_type_hint()) {
+			case access_token: // retrieve the refresh token ID from the JWT
+				String refreshTokenId = this.getRefreshTokenId(revocationRequest.getToken());
+				if (refreshTokenId == null) {
+					// Access tokens that were not issued alongside refresh tokens cannot be revoked.
+					throw new IllegalArgumentException("The access token was not issued via a refresh token. It cannot be revoked.");
+				}
+				// Revoke via token doesn't require authorization, so we use the admin user
+				UserInfo adminUserInfo = userManager.getUserInfo(AuthorizationConstants.BOOTSTRAP_PRINCIPAL.THE_ADMIN_USER.getPrincipalId());
+				oauthRefreshTokenManager.revokeRefreshToken(adminUserInfo, refreshTokenId);
```

Comment:
please add a method to OauthRefreshTokenManager:  revokeRefreshToken(String authenticateClientId, String refreshTokenId)
which will work if the clientId matches the client in the token.